packages/flutter_tools/lib/src/globals_null_migrated.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import 'base/config.dart';
+import 'base/context.dart';
+import 'base/error_handling_io.dart';
+import 'base/file_system.dart';
+import 'base/io.dart';
+import 'base/logger.dart';
+import 'base/net.dart';
+import 'base/os.dart';
+import 'base/platform.dart';
+import 'base/process.dart';
+import 'base/signals.dart';
+import 'base/template.dart';
+import 'base/terminal.dart';
+import 'base/time.dart';
+import 'base/user_messages.dart';
+import 'ios/plist_parser.dart';
+import 'isolated/mustache_template.dart';
+
+Config get config => context.get<Config>() ?? _config;
+final Config _config = Config(
```

Comment:
These default fallbacks are new.

Reply:
I could also instantiate these lazily as `plistParser` does below:
```dart
Config get config => context.get<Config>() ?? (
  _config ??= Config(
    Config.kFlutterSettings,
    fileSystem: fs,
    logger: logger,
    platform: platform,
  ));
Config? _config;
```

Reply:
I would be worried about parts of the tool using these defaults "for real" - For example, the real logger setup logic is much more complicated.

Could we instead add more defaults for testUsingContext? That should at least limit the splash damage if something goes wrong.

Reply:
They pass fine in `testUsingContext`, the problem is when they are being set outside a context, like in `setUp`.  Like this:
```
  setUp(() async {
...
    projectUnderTest = FlutterProject.fromDirectory(fileSystem.directory('project'));
  });
...
    testWithoutContext('creates objective-c Podfile when not present', () async {
      await cocoaPodsUnderTest.setupPodfile(projectUnderTest.ios);
```

Let me try a more focused change...

Reply:
Right.

I guess ... we could make them nullable for now?

Reply:
Or https://source.corp.google.com/piper///depot/google3/mobile/flutter/cli/test/context_test.dart;l=182
```
00:00 +7 -1: Google3DeviceManager: Honors a specified project platform [E]                                                                                                                             
  Null check operator used on a null value
  package:flutter_tools/src/globals_null_migrated.dart 26:43   logger
  package:mobile.flutter.cli/context.dart 619:29               new Google3DeviceManager
  google3:///mobile/flutter/cli/test/context_test.dart 182:11  main.<fn>.<fn>
```